website/docs/r/api_gateway_authorizer.html.markdown

```diff
@@ -113,3 +113,4 @@ The following arguments are supported:
 	For `TOKEN` type, this value should be a regular expression. The incoming token from the client is matched
 	against this expression, and will proceed if the token matches. If the token doesn't match,
 	the client receives a 401 Unauthorized response.
+* `provider_arns` - (Optional, required for type `COGNITO_USER_POOLS`) A list of the Amazon Cognito user pool ARNs. Each element is of this format: `arn:aws:cognito-idp:{region}:{account_id}:userpool/{user_pool_id}`.
```

Comment:
Would you mind taking a look at adding a `CustomizeDiff` function that adds plan-time validation for this situation? Maybe something similar to `aws_dynamodb_table` function can work here too:

https://github.com/terraform-providers/terraform-provider-aws/blob/0cd8e5a0e6572740511f54c44c483a0549ecee2e/aws/validators.go#L2236